plugins/org.python.pydev/tests/org/python/pydev/editor/PyAutoIndentStrategyTest.java

```diff
@@ -1297,6 +1297,15 @@ public void testAutoSelf2() {
         assertEquals(expected, docCmd.text);
     }
 
+    public void testExitAutoSelf() {
+        strategy.setIndentPrefs(new TestIndentPrefs(false, 4, true));
+        String doc = "def __exit__";
```

Comment:
This isn't correct. The `__exit__` should only be completed that way when it's inside a class (so, in this case it should just complete with `():`.

Reply:
Alright, I did not know that